internal/database/db.go

```diff
@@ -465,3 +468,95 @@ func (dbs *SQLdb) Close() {
 	db := dbs.DB
 	db.Close()
 }
+
+// DisableFile marks the file as 'DISABLE'
+func (dbs *SQLdb) DisableFile(filePath, user string) error {
+	var err error
+	for count := 1; count <= dbRetryTimes; count++ {
+		err = dbs.disableFile(filePath, user)
+		if err == nil {
+			break
+		}
+	}
+
+	return err
+}
+func (dbs *SQLdb) disableFile(filePath, user string) error {
+	dbs.checkAndReconnectIfNeeded()
+
+	db := dbs.DB
+	const disable = "UPDATE local_ega.files SET status = 'DISABLED' WHERE inbox_path = $1 and elixir_id = $2 and status != 'READY';"
+	result, err := db.Exec(disable, filePath, user)
+	if err != nil {
+		return err
+	}
+	if rowsAffected, _ := result.RowsAffected(); rowsAffected == 0 {
+		return errors.New("something went wrong with the query zero rows were changed")
+	}
+
+	return nil
+}
+
+// GetStatus returnes the status for a given fileID
+func (dbs *SQLdb) GetStatus(fileID int64) (string, error) {
+	status, err := dbs.getStatus(fileID)
+	if err != nil {
+		return "", err
+	}
+
+	return status, nil
+}
+func (dbs *SQLdb) getStatus(fileID int64) (string, error) {
+	dbs.checkAndReconnectIfNeeded()
+
+	db := dbs.DB
+	const check = "SELECT status FROM local_ega.files WHERE id = $1;"
+	var status string
+	if err := db.QueryRow(check, fileID).Scan(&status); err != nil {
+		return "", err
+	}
+
+	return status, nil
+}
+
+// GetFileID returnes the ID for a given inbox path and user combination
+func (dbs *SQLdb) GetFileID(filePath, user string) (int64, error) {
+	fileID, err := dbs.getFileID(filePath, user)
+	if err != nil {
+		return 0, err
+	}
+
+	return fileID, nil
+}
+func (dbs *SQLdb) getFileID(filePath, user string) (int64, error) {
+	dbs.checkAndReconnectIfNeeded()
+
+	db := dbs.DB
+	const check = "SELECT id FROM local_ega.files WHERE inbox_path = $1 and elixir_id = $2;"
```

Comment:
There is no guarantee that this is a single unique file. In the case where the same user uploads a README for two projects this will match both of them... you probably need to add `archive_path` to make this unique

Reply:
Well, that problem is not something we can do anything about since the message that comes from CEGA doesn't have an archive path for a file that is not ingested yet.